src/main/java/com/howtech/posstorageapi/controllers/PosStorageApiController.java

```diff
@@ -0,0 +1,29 @@
+package com.howtech.posstorageapi.controllers;
+
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.multipart.MultipartFile;
+
+@RestController
+@RequestMapping("/storage")
+public class PosStorageApiController {
+
+
+
+
+    /**
+     * Storage upload and download image url
+     */
+
+    /**
+     * @param file a file to upload
+     * @return the path to the file
+     * @throws
+     */
+    @PostMapping("/store-profile/{store_id}")
+    public String uploadStorageLogo(@PathVariable (name = "store_id") Long storeId,
+                                    @RequestPart(value = "file")MultipartFile file)) {
```

Comment:
We're gonna have to finish this one first, also just looked at the architecture and it will have to be modified slightly. Since I took it out it will now need a persistant key store to map the returned URL with the s3 url so that we can save it and then return the s3 file back through the get request. I can show you how it will look and show you how to dertermine the DTOs you will need today.